package/kedro_viz/models/flowchart.py

```diff
@@ -61,54 +69,72 @@ class GraphNodeType(str, Enum):
     )
 
 
-@dataclass(frozen=True)
-class ModularPipelineChild:
-    """Represent a child of a modular pipeline."""
+class ModularPipelineChild(BaseModel, frozen=True):
+    """Represent a child of a modular pipeline.
+
+    Args:
+        id (str): Id of the modular pipeline child
+        type (GraphNodeType): Type of modular pipeline child
+    """
 
     id: str
     type: GraphNodeType
 
+    def __hash__(self) -> int:
+        return hash(self.id)
+
```

Comment:
Why this is needed now? Is this a bug fix or part of the refactoring

Reply:
Dataclasses provide default methods for example (__hash__), but for pydantic base class, we need to add these methods manually

Reply:
my bad, This was added in the earlier refactor when I did not add frozen attribute which made the classes unhashable type when used in sets and dict. Now this is not required anymore. Thanks for pointing

Reply:
looks like we need the __hash__ function as pylint does not understand pydantic frozen class. Though it does not complain with frozen dataclass - [issue]( https://app.circleci.com/pipelines/github/kedro-org/kedro-viz/10998/workflows/21c09563-e6ce-4cf5-84da-1505b416a428/jobs/66399)

Reply:
is this only a pylint issue? if so, can we ignore it?

Reply:
yes according to https://github.com/pydantic/pydantic/pull/1881 , we do not need to add __hash__ it is generated for frozen class. Will ignore pylint issue

Reply:
removed hash function